Content.Client/DoAfter/DoAfterOverlay.cs

```diff
@@ -102,7 +105,7 @@ protected override void Draw(in OverlayDrawArgs args)
             {
                 // Hide some DoAfters from other players for stealthy actions (ie: thieving gloves)
                 var alpha = 1f;
-                if (doAfter.Args.Hidden)
+                if (doAfter.Args.Hidden || _container.IsEntityInContainer(uid))
                 {
```

Comment:
This is going to be checked inside of the most expensive loop when you should be checking it above + also I believe you can pass metadatacomponent in and more.